src/System.Net.HttpListener/src/System/Net/HttpListenerRequest.cs

```diff
@@ -44,6 +47,68 @@ public string[] UserLanguages
             }
         }
 
+        private static MethodInfo s_internalAddMethod = null;
+        private static MethodInfo InternalAddMethod
+        {
+            get
+            {
+                if (s_internalAddMethod == null)
+                {
+                    // We need to use CookieCollection.InternalAdd, as this method performs no validation on the Cookies.
+                    // Unfortunately this API is internal so we use reflection to access it. The method is cached for performance reasons.
+                    MethodInfo method = typeof(CookieCollection).GetMethod("InternalAdd", BindingFlags.NonPublic | BindingFlags.Instance);
+                    Debug.Assert(method != null, "We need to use an internal method named InternalAdd that is declared on Cookie.");
+                    s_internalAddMethod = method;
+                }
+
+                return s_internalAddMethod;
+            }
+        }
+
+        private CookieCollection ParseCookies(Uri uri, string setCookieHeader)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Info(this, "uri:" + uri + " setCookieHeader:" + setCookieHeader);
+            CookieCollection cookies = new CookieCollection();
+            CookieParser parser = new CookieParser(setCookieHeader);
+            for (;;)
+            {
+                Cookie cookie = parser.GetServer();
+                if (NetEventSource.IsEnabled) NetEventSource.Info(this, "CookieParser returned cookie:" + cookie?.ToString());
```

Comment:
nit: space after "CookieParser returned cookie:".
I think the logging doesn't make sense if the cookie is null as the string then will be "CookieParser returned cookie:" //EOL. I would move it after the cookie null check